### PR TITLE
Bulk update of Select to EntityRef for FKs

### PR DESCRIPTION
**Pull request by aydun**
Overview
----------------------------------------
Generally 'EntityRef' is preferred to 'Select' for Foreign Keys.  This updates them.

See [this chat](https://chat.civicrm.org/civicrm/pl/co7oym8g9fr7jmaer83c1umrfa)

Before
----------------------------------------
`<html><type>Select</type></html>` used for some Foreign Keys

After
----------------------------------------
`<html><type>EntityRef</type></html>` used for fewer Foreign Keys.  As advised by @colemanw in the chat linked above, this does not change fields used by `CRM_Core_Form::addField()`

Technical Details
----------------------------------------
The html type is mostly used by FormBuilder so this has little if any impact to core.

Comments
----------------------------------------

### CODE REVIEW

**Comment by colemanw** [comment]
I'm worried this may have unintended consequences. In the chat I noted that `CRM_Core_Form::addField` *does* use this flag, so this would change any core form that uses `addField` with any of these selects.
Also, in cases where the option list is always very small, a select offers better UX because there's no ajax latency in loading options when the user clicks on it. Examples: `payment_processor_id`, `sms_provider_id`, `domain_id, `component_id`...
Also there are some whose options get restricted in other ways, like `Address.country_id` which follows a sitewide limit on available countries.
So long story short, I think this kind of thing is best done on a per-entity basis with some care rather than as a bulk update.

**Comment by aydun** [comment]
This excludes any that are used with `addField` but I hear the concern.
